src/client/pages/InkomenDetail/InkomenDetailBbz.tsx

```diff
@@ -42,10 +46,52 @@ export default function InkomenDetailBbz() {
       highlightKey={false}
       statusLabel={() => `Bbz-aanvraag`}
       showStatusLineConnection={!hasDecisionStep} // There is no logical connection between the historic BBZ steps, therefor we do not show the checkmarks as they imply a linear proces.
-      reverseSteps={hasDecisionStep} // For an unknown business reason, the historic steps of BBZ are shown in reverse.
       documentPathForTracking={(document) =>
         `/downloads/inkomen/bbz/${document.title.split(/\n/)[0]}`
       }
+      getItems={() => {
+        if (!statusItem) {
+          return [];
+        }
+
+        if (
+          statusItem.steps.some((s) =>
+            REQUEST_PROCESS_COMPLETED_STATUS_IDS.includes(s.id)
+          )
+        ) {
+          // If its a completed request, pass data without processing (show's all steps seperate)
+          return [statusItem] as any[];
+        }
+
+        // It's an open request, combine all aanvraag steps into one.
+        const requests = statusItem.steps.filter((s) => s.id === 'aanvraag');
+
+        // Only 1 aanvraag steps, early return
+        if (requests.length === 1) {
+          return [statusItem] as any[];
+        }
+
+        const requestStep = { ...requests[0] };
+        const otherRequestDocuments = requests
+          .slice(1)
+          .map((s) => s.documents)
+          .flat();
+
+        // Combine the documents
+        requestStep.documents = [
+          ...requestStep.documents,
+          ...otherRequestDocuments,
+        ];
+        const otherSteps = statusItem.steps.filter((s) => s.id !== 'aanvraag');
+
+        // Create new item.
+        const returnItem = {
+          ...statusItem,
+          steps: [requestStep, ...otherSteps],
+        };
+
+        return [returnItem] as any[];
+      }}

```

Comment:
Deze logica hoort niet in de front-end. Ik heb een pr gemaakt https://github.com/Amsterdam/mijn-wpi/pull/42/files